pkg/liqoctl/status/pods.go

```diff
@@ -0,0 +1,397 @@
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/liqotech/liqo/pkg/utils/slice"
+)
+
+const (
+	checkerName = "liqo-control-plane"
+)
+
+var (
+	liqoDeployments = []string{
+		"capsule-controller-manager",
+		"liqo-controller-manager",
+		"liqo-network-manager",
+		"liqo-crd-replicator",
+		"liqo-webhook",
+		"liqo-gateway",
+		"liqo-auth",
+	}
+	liqoDaemonSets = []string{
+		"liqo-route",
+	}
+)
+
+// errorCount is a struct holding a list of errors.
+type errorCount struct {
+	errors []error
+}
+
+// collectionError struct holding the error for a given deployment of Liqo while collecting its status.
+type collectionError struct {
+	appType string
+	appName string
+	err     error
+}
+
+// newCollectionError return a new collectionError with the given arguments set.
+func newCollectionError(appType, appName string, err error) collectionError {
+	return collectionError{
+		appType: appType,
+		appName: appName,
+		err:     err,
+	}
+}
+
+// errorCountMap is a map with:
+// key -> name of the pod;
+// value -> errorCount for the given pod.
+type errorCountMap map[string]*errorCount
+
+// podState counts the number of pods in a particular state for a given deployment.
+type podState struct {
+	// errorFlag set to true if no errors are present.
+	errorFlag bool
+
+	// deploymentType is the type of deployment ("Deployment", "DaemonSet", ...).
+	deploymentType string
+
+	// desired is the number of desired pods to be scheduled.
+	desired int
+
+	// ready is the number of ready pods.
+	ready int
+
+	// available is the number of available pods.
+	available int
+
+	// unavailable is the number of unavailable pods.
+	unavailable int
+
+	// imageVersions is the name of the images used by the pods (containers/init-containers), denoting the version of the images.
+	imageVersions []string
+
+	// errors is the aggregated errors of all pods.
+	errors errorCountMap
+}
+
+// podStateMap is a map with:
+// key -> name of the Liqo component;
+// value -> podState.
+type podStateMap map[string]podState
+
+func newPodState(dType string) podState {
+	return podState{
+		deploymentType: dType,
+	}
+}
+
+// getImages returns the images used by the current deployment/application.
+func (ps *podState) getImages() []string {
+	return ps.imageVersions
+}
+
+// setImages sets the images passed as argument.
+func (ps *podState) setImages(images []string) {
+	ps.imageVersions = images
+}
+
+// addImageVersion adds an image version to the existing ones.
+func (ps *podState) addImageVersion(imageVersion string) {
+	iv := ps.getImages()
+	if !(slice.ContainsString(iv, imageVersion)) {
+		iv = append(iv, imageVersion)
+	}
+	ps.setImages(iv)
+}
+
+// getErrorCount returns the errorCount for given pod belonging to the current deployment/application.
+func (ps *podState) getErrorCount(pod string) *errorCount {
+	var errors errorCountMap
+	// Get the errorCountMap for the current deployment.
+	if ps.errors == nil {
+		ps.errors = errorCountMap{}
+	}
+	errors = ps.errors
```

Comment:
Could be initialized by the NewPodState function, to avoid handling this case here.